varnish.m4

```diff
@@ -201,7 +201,15 @@ AC_DEFUN([_VARNISH_VMOD], [
 	dnl Define the VCL directory for automake
 	AC_SUBST([vmod_$1_vcldir], [\${vcldir}/$1])
 
-	VMOD_RULES="
+	m4_case([$2],
+		[static],
+		[m4_pushdef([VCC_SRC], [\$(srcdir)/vmod_$1.vcc])],
+		[generated],
+		[m4_pushdef([VCC_SRC],
+			[\$(builddir)/vmod_$1.vcc \$(srcdir)/vmod_$1.vcc])],
```

Comment:
this would check for both locations while, when the semantics are `generated`, we would only need to look in `$(builddir)`.
This way, the change would, in my mind, be fully backwards compatible with the previous vmodtool and we would not need to tie merging this change to the release, or do I miss something?

Reply:
We had this discussion a couple weeks ago but I'll summarize it here: this wouldn't work in a project that produce autotools dist archives that are meant to support multiple Varnish versions. So a new macro was added and documented to only work with Varnish 7, since vmodtool would not understand the VPATH trick with older releases.

Reply:
You are right about compatibility with existing releases, and we avoid any issues in this area with a new macro.
What I am asking is this: For `generated`, the vcc can be expected in `$(builddir)` always. So I think we do not need the second vmodtool argument.

Reply:
Not if you ship it in the dist archive, downstream to `make dist` it belongs in `$(srcdir)`.

Reply:
ah, I see. thank you!

Reply:
> We had this discussion a couple weeks ago

;-)

Reply:
I apologize for my bad memory, my bad spelling and the fact that I should on public mailing lists when someone mentions water transport systems in conjunction with mollusk exoskeletons and a particular shape of hair.